### ISSUE TITLE
File Explorer built into the arranger.

### ISSUE THREAD

**Issue by Gabrielxd195** [muse-sequencer/muse#1023]
I do not know what future plans they have for MusE, but a file explorer that is integrated in the arranger is missing to be able to access your files quickly, since currently you can import audio, midi and open projects from the files tab, but it is very impractical to do it this way, especially when you have to import many files to the project. an ideal file explorer might have the following functions:

1- Import audio and midi samples.
2- Play the samples with the keyboard arrows.
3- Open projects quickly.
4- Import soundfonts.
5- Even have a tab to import plugins.
6- Mark paths, directories and files as favorites.

To give an idea of ​​what I explain I show you the LMMS and Zrythm file explorer as an example.

![Explorador de lmms edit](https://user-images.githubusercontent.com/26127693/120563848-8e6fab00-c3d7-11eb-8642-158c57858abd.jpg)
The lmms file browser plays samples with keyboard arrows and imports projects quickly.

![Explorador de zrythm edit](https://user-images.githubusercontent.com/26127693/120563863-9cbdc700-c3d7-11eb-9f2a-576739b90457.png)
The Zrythm explorer divides categories into tabs, and even imports plugins and soundfonts.

I don't know how much support the muse team will have to implement this function, but at least the idea is there, and something like that would greatly improve the MusE workflow, since many users do not record their songs, but compose them base of imported samples. Blessings to all.

**Comment by terminator356** [comment]
I agree about having a window containing 'resources' such as wave/midi files, wave/midi clips, favorite clips etc.

For that reason, I have a branch where I am attempting to do just that. It is in its infancy though.
In the Arranger there is a menu item 'Clip List'. It has never been very useful, needing some attention.
For now it simply shows a read-only list of wave files that are used in the song.
I want to rock it up with drag-able clips, resources and so on.
Drag 'em back and forth to and from the canvases and so on.
(Technical note: In thought, at least, dragging things _off of_ a canvas is proving difficult to solve
 because the canvas wants to auto-scroll rather than letting items be dragged off of it ! I have a few ideas...)

The branch is lagging. So many things keep popping up it's hard to maintain. Hopefully soon I can get back to it.

[ Psst! It's got a few goodies that I would really like to cherry-pick while I wait.
For example all the numeric time readouts and entry boxes and the BigTime window can change their time format:
BBT, Ticks, HMSF, Frames, and Transport time code (similar to HMSF but it will read the same as
 QJackCtl's readout, not our current slightly incompatible HMSF format.) ]
